### ISSUE TITLE
How should I finetune the parameters?

### ISSUE THREAD

**Issue by whu-lyh** [zhanjiawang/VPFBR-L#1]
Thanks for your job. I test your registration method on my own point cloud data which is collected from a MLS sensor. The result I got is bad compared to the raw relative pose. The result is shown as followings. The red point cloud is the result of your method, while the blue and green are raw point cloud pair.
![](https://cdn.jsdelivr.net/gh/whu-lyh/images@latest/img/image-20230711123258963.png)
I wander how should I finetune the parameters to get a good result.

**Comment by whu-lyh** [comment]
@zhanjiawang

**Comment by zhanjiawang** [comment]
This method requires at least three pairs of corresponding planes to determine the transformation matrix. Is it possible that the poor registration effect of point clouds is due to the insufficient number of planar features in the scene. If the number of planes is too small, this method may not work properly. You can try reducing face_voxel_size in the algorithm and increasing curvature_threshold and select_plane_number to see if you can achieve better registration results, but these adjustments may increase registration time. In addition, as this method belongs to coarse registration, it may only obtain approximate solutions of the transformation matrix. You can try adding a fine registration method after the registration result to obtain better results.

**Comment by whu-lyh** [comment]
Thanks, jiawang. I will try to adjust these parameters. The road contributes much to planes, and the other plane-type point cloud is either a small region or vertically distributed. Will these cause the failure of your method?
In fact, I have tried several fine registration methods but none of them works, cause the scene in my point cloud contains pedestrians and vehicles. And the point cloud is collected under an overhead bridge, which is normal in urban scenarios.

**Comment by ronan222222** [comment]
> 谢谢你的工作。我使用我自己的点云数据测试您的注册方法，该数据是从 MLS 传感器收集的。与原始相对姿势相比，我得到的结果很糟糕。结果如下所示。红色点云是您的方法的结果，而蓝色和绿色是原始点云对。 ![](https://camo.githubusercontent.com/98e3f9aa673feaa626df040387517fe2d815557d5e643c090561af89c7f6d06a/68747470733a2f2f63646e2e6a7364656c6976722e6e65742f67682f7768752d6c79682f696d61676573406c61746573742f696d672f696d6167652d32303233303731313132333235383936332e706e67)我徘徊在我应该如何微调参数以获得好结果。

你好，我想问一下您是怎么复现的呢？我不知道应该如何安装这几个库，是不是下载完这几个库之后，还要修改一下配置文件呀？

**Comment by whu-lyh** [comment]
@ronan222222, Yes， I just configured this repo based on my local developing environments. There are less code comments, and I failed to fine-tune the parameters.

**Comment by ronan222222** [comment]
> @ronan222222, Yes， I just configured this repo based on my local developing environments. There are less code comments, and I failed to fine-tune the parameters.

Thank you very much for your reply. Recently, I have been reading the registration article, and now I am thinking of making a prize to place two radars at the two ends of a long straight road section, and then register the point cloud generated by them, so as to expand the detection range of LiDAR. However, I have been puzzled about the small overlap of my point clouds, and I do not know whether I can use the registration method. Can you give me some advice?

**Comment by whu-lyh** [comment]
I don't think this matching method could solve your problem especially the small overlap between two point clouds. The small overlap ratio is a hard scenario to handle..

**Comment by ronan222222** [comment]
> I don't think this matching method could solve your problem especially the small overlap between two point clouds. The small overlap ratio is a hard scenario to handle..

Thank you for your answer, I will look for other ways。
